packages/vaex-core/vaex/expression.py

```diff
@@ -276,6 +276,61 @@ def max(self, binby=[], limits=None, shape=default_shape, selection=False, delay
         kwargs['expression'] = self.expression
         return self.ds.max(**kwargs)
 
+    def value_counts(self, dropna=False, ascending=False):
+        """Computes counts of unique values
+         WARNING:
+          * If the expression/column is not categorical, it will be converted on the fly
+          * dropna is False by default, it is True by default in pandas
+         :param dropna: when True, it will not report the missing values
+        :param ascending: when False (default) it will report the most frequent occuring item first
+        :returns: Pandas series containing the counts
+        """
+        from pandas import Series
+        df = self.ds
+        # whatever we do, it's gonna be similar to ds.categorise, so better
+        # to use that path
+        if not df.iscategory(self.expression):
+            df = df.copy()
+            # consider using label_encode.. which seems like a wrong name actually
+            # ds.categorize(self.expression)
+            df = df.label_encode(self.expression)
+        N = df.category_count(self.expression)
+        raw_counts = df.count(binby=self.expression, edges=True, limits=[-0.5, N-0.5], shape=N)
+        assert raw_counts[1] == 0, "unexpected data outside of limits"
+        assert raw_counts[-1] == 0, "unexpected data outside of limits"
+        if dropna:
+            counts = raw_counts[2:-1]
+            index = df.category_values(self.expression)
+        else:
+            # first element contains missing values/wrong values
+            counts = np.zeros(N+1, dtype=np.int64)
+            counts[0] = raw_counts[0]
+            counts[1:] = raw_counts[2:-1]
+            if df.category_values(self.expression).dtype.kind == 'f':
+                index = [np.nan] + df.category_values(self.expression).tolist()
+            else:
+                index = ["missing"] + df.category_labels(self.expression)
+        order = np.argsort(counts)
+        index = np.asanyarray(index)
+        if not ascending:
+            order = order[::-1]
+        counts = counts[order]
+        index = index[order]
+        # filter out values not present (which can happen due to how categorize works)
+        ok = counts > 0
+        if dropna and df.category_values(self.expression).dtype.kind == 'f':
+            nan_mask = np.isnan(index)
+            if np.any(nan_mask):
+                ok = ok & ~nan_mask
+        if np.ma.isMaskedArray(index):
+            ok = ok & ~np.ma.getmaskarray(index)
+        counts = counts[ok]
+        index = index[ok]
+        return Series(counts, index=index)
+
+    def unique(self):
+        return self.ds.unique(self.expression)
```

Comment:
Should this be also renamed to `self.df.unique` at this point?

Reply:
yes, lets do that separately though, not in this PR.